lib/medical_records/bb_internal/client.rb

```diff
@@ -303,6 +285,42 @@ def get_study_id_from_cache(id)
       end
     end
 
+    ##
+    # Takes an array of objects with a studyIdUrn field. For each object, if the studyIdUrn is already mapped to a
+    # cached UUID in redis, simply replace all the studyIdUrn with its mapped value. If the value is not yet mapped
+    # in redis, create the mapped UUID first, then replace the studyIdUrn in the object. This is to prevent the
+    # study_id from being exposed to the client.
+    #
+    # @param [Array] data - An array which contains objects that have a studyIdUrn field.
+    #
+    # @return [Array] A modified array in which all the objects have has studyIdUrn mapped to a UUID.
+    #
+    def map_study_ids(data)
+      study_data_cached = get_study_data_from_cache
+      study_data_hash = JSON.parse(study_data_cached) if study_data_cached
+      id_uuid_map = study_data_hash || {}
+
+      modified_data = data.map do |obj|
+        study_id = obj['studyIdUrn']
+
+        existing_uuid, = study_data_hash.find { |_, v| v == study_id.to_s } if study_data_hash
```

Comment:
This should return the key that matches the value (`study_id`) and do a `nil` check with the safe navigation operator.

```suggestion
        existing_uuid = study_data_hash&.key(study_id.to_s)
```